btree/btree.c

```diff
@@ -8372,24 +8371,22 @@ M0_INTERNAL int m0_btree_lrulist_purge(int64_t size)
 	node = ndlist_tlist_tail(&btree_lru_nds);
 	while (node != NULL && size > 0) {
 		curr_size = 0;
-		prev = ndlist_tlist_prev(&btree_lru_nds, node);
+		prev      = ndlist_tlist_prev(&btree_lru_nds, node);
 		if (node->n_txref == 0 && node->n_ref == 0) {
 			curr_size = node->n_size;
-			seg = node->n_tree->t_seg;
-			rnode = segaddr_addr(&node->n_addr);
+			seg       = node->n_tree->t_seg;
+			rnode     = segaddr_addr(&node->n_addr);
 			ndlist_tlink_del_fini(node);
```

Comment:
The node needs to be removed from the list only if the remap is successful. If not then the node needs to stay in the list.

Reply:
Done